Larger improvements

apeltzer opened the pull request:
First of all, thank you @spaul91 and team to have this open sourced - this already helped us a lot to get to some MVP for ingesting TSO500 outputs more quickly 🥇 

We have added quite a bit of additional functionality and wanted to contribute that for your project too - see this PR for a list of changes / updates and additions to documentation, code and parsing functionality / filtering possibilities. We hoep you also find this useful :)

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/eastgenomics/TSO500R/5)
<!-- Reviewable:end -->

spaul91 committed:
ignore files

spaul91 committed:
suppress useless warning

spaul91 committed:
bugfixes

spaul91 committed:
RMD workbook

spaul91 committed:
updated filtering to remove variants with unwanted consequences, instead of include only those with needed consequences

spaul91 committed:
updated documentation

spaul91 committed:
Merge pull request #2 from eastgenomics/issue_1

Issue #1 fix:  Variant consequence parsing removes required variants

apeltzer committed:
Update README.md

Some cleanup of readme

apeltzer committed:
Merge pull request #1 from apeltzer/develop

Bring master branch up to speed

christopher-mohr committed:
Insert NA for empty columns in variant table, fix NA check

christopher-mohr committed:
adapt method name to functionality

christopher-mohr committed:
add methods for extracting information on gene amplifications, fusions, splice variants, update doc using roxygen2

apeltzer committed:
Merge pull request #2 from christopher-mohr/development

Extend functionality

christopher-mohr committed:
add functionality for reading tmb trace and filtering

apeltzer committed:
Merge pull request #3 from christopher-mohr/development

Add functionality for TMB trace files

christopher-mohr committed:
add plotting and exporting functionlity

apeltzer committed:
Merge pull request #4 from christopher-mohr/development

Add plotting and exporting functionlity

christopher-mohr committed:
add possibility to add additional annotations to OncoPrint

christopher-mohr committed:
add functionaltiy to parse and access quality metrics provided in MetricsOutput.tsv

christopher-mohr committed:
add functionality to parse detailed tmb information given in *tmb.json files

christopher-mohr committed:
improve parser for MetricsOutput to avoid columns without content

apeltzer committed:
Merge pull request #5 from christopher-mohr/development

Improve plotting, add parsers

christopher-mohr committed:
make parser for QC and tmb details DRAGEN-compatible

apeltzer committed:
Merge pull request #6 from christopher-mohr/development

Improve QC and TMB details parser

christopher-mohr committed:
rename functions to prevent mixup

apeltzer committed:
Merge pull request #7 from christopher-mohr/development

Rename functions to prevent mixup

christopher-mohr committed:
remove code that deletes last column of matrix

apeltzer committed:
Merge pull request #8 from christopher-mohr/development

christopher-mohr committed:
add function to generate DRAGEN analysis pipeline sample sheet

christopher-mohr committed:
update documentation

christopher-mohr committed:
Merge branch 'development' of https://github.com/christopher-mohr/TSO500R into development

apeltzer committed:
Merge pull request #9 from christopher-mohr/development

Add functionality to generate DRAGEN Analysis Pipeline Sample Sheet

christopher-mohr committed:
improve string match in parser

apeltzer committed:
Merge pull request #10 from christopher-mohr/development

Improve string matching in parser

christopher-mohr committed:
replace hardcoded indices, add local_app variable for cvo data

christopher-mohr committed:
change variable names

christopher-mohr committed:
remove print

christopher-mohr committed:
change naming of quality metrics objects

christopher-mohr committed:
add documentation

apeltzer committed:
Merge pull request #12 from christopher-mohr/development

Add documentation, improve parsers

christopher-mohr committed:
add missing roxygen export statements and add missing info

apeltzer committed:
Merge pull request #13 from christopher-mohr/development

Roxygen improvements

spaul91 commented:
Hi @apeltzer ,

Thank you for the PR, very pleased to hear this module is useful :) I'll review your code and let you know if anything needs updating before merging.

Many thanks,

Sophie Paul

christopher-mohr committed:
update package DESCRIPTION

christopher-mohr committed:
update LICENSE file

apeltzer committed:
Merge pull request #14 from christopher-mohr/development

Update DESCRIPTION and LICENSE

christopher-mohr committed:
add indent to allow multiple lines

apeltzer committed:
Merge pull request #15 from christopher-mohr/development

Add indent to allow multiple lines

christopher-mohr committed:
fix issue with DESCRIPTION, adapt roles

christopher-mohr committed:
fix remaining documentation issues

apeltzer committed:
Merge pull request #16 from christopher-mohr/development

Fix remaining documentation issues

christopher-mohr committed:
transform NA strings to NA to avoid warnings

christopher-mohr committed:
avoid warnings in nested list

christopher-mohr committed:
add functionality for reading raw CNV data

christopher-mohr committed:
add pattern for non-localapp CNV files

apeltzer committed:
Merge pull request #17 from christopher-mohr/development

Add functionality for parsing raw CNV data

christopher-mohr committed:
extend cnv file pattern for read_cnv_data function

christopher-mohr committed:
Merge branch 'apeltzer:main' into development

apeltzer committed:
Merge pull request #18 from christopher-mohr/development

Extend cnv file pattern for read_cnv_data function

grst committed:
Update write.R

The column `Pair_ID` doesn't exist in the samplesheet, however a `Pat_ID` (patientID) exists. 

Even if it existed, it would get overwritten two lines further down in the `mutate` statement.

grst committed:
Search for CVO files recusively

Useful for specifying a DRAGEN output folder, where CombinedVariantOutput.tsv files are organized in subfolders by sample.

grst committed:
Add function to generate QC metric tables with gt

grst committed:
Update tmb.R

grst committed:
Update cnv.R

grst committed:
Merge pull request #19 from grst/patch-1

Fix create_samplesheet

grst committed:
Merge pull request #20 from grst/patch-2

Search for CVO/TMBtrace files recusively

grst committed:
Merge pull request #21 from grst/patch-3

Add function to generate QC metric tables with gt

christopher-mohr committed:
update NAMESPACE, docs

christopher-mohr committed:
Merge pull request #24 from christopher-mohr/update_namespace

Update NAMESPACE, docs

christopher-mohr committed:
fix importFrom statement

christopher-mohr committed:
Merge pull request #25 from christopher-mohr/fix_namespace

Fix importFrom statement

christopher-mohr committed:
make CVO and tmb trace parser compatible with ctdna output

christopher-mohr committed:
adapt getter functions

christopher-mohr committed:
make metricsoutput parser compatible with ctdna pipeline

christopher-mohr committed:
add ctdna to docs

christopher-mohr committed:
Merge branch 'main' into add_ctdna_parser

christopher-mohr committed:
update docs

christopher-mohr committed:
use predefined strings in parser

christopher-mohr committed:
Merge pull request #26 from christopher-mohr/add_ctdna_parser

Add functionality for ctdna pipeline output

christopher-mohr committed:
add parameter to dragen samplesheet function

christopher-mohr committed:
use param directly

apeltzer committed:
Merge pull request #27 from christopher-mohr/fix_samplesheet_parser

Add parameter to dragen samplesheet function

apeltzer committed:
Bugfix for DVC 

This bugfix handles issues with dvc files. The regex has been extended to specifically look for output files that end with *.tsv ($ ensures this is the end of the string in the pattern). 

This resolves breaking with *.tsv.dvc files created when using data version control. Might be required in other cases too, will give it a try.

apeltzer committed:
Fix for DVC and TMB

Also fix for TMB data

apeltzer committed:
Fix for CNV

apeltzer committed:
Fix for QMO and dvc

christopher-mohr committed:
Merge pull request #28 from Boehringer-Ingelheim/fix-for-dvc

Bugfix for DVC

christopher-mohr committed:
fix qc table for contamination p-value

christopher-mohr committed:
minor fix in README

christopher-mohr committed:
remove the use of dTMCP package

christopher-mohr committed:
Merge pull request #30 from christopher-mohr/fix_qc_table

Fix qc table for contamination p-value

christopher-mohr committed:
make tmb and msi parsing safer

apeltzer committed:
Merge pull request #32 from christopher-mohr/fix_sample_id_parser

Make tmb and msi parsing safer

christopher-mohr committed:
define separate function for parsing illumina sample sheets

christopher-mohr committed:
improve docs

christopher-mohr committed:
add github test workflow, docs

christopher-mohr committed:
Merge branch 'main' into ci_doc_website

christopher-mohr committed:
use variable for column name

christopher-mohr committed:
add action for docs build

christopher-mohr committed:
update docs

christopher-mohr committed:
fix variable names in docs

christopher-mohr committed:
small docs changes

christopher-mohr committed:
add actions to workflows subdir

christopher-mohr committed:
update NAMESPACE

christopher-mohr committed:
update DESCRIPTION file

christopher-mohr committed:
fix imports and warnings

christopher-mohr committed:
fix issue in Rd file

christopher-mohr committed:
add lintr config, fix rmd check warnings

christopher-mohr committed:
fix more warnings

christopher-mohr committed:
fix lintr warnings

christopher-mohr committed:
fix issues with namespace

christopher-mohr committed:
resolve warnings

christopher-mohr committed:
update imports

christopher-mohr committed:
more warnings

christopher-mohr committed:
update imports

christopher-mohr committed:
add rlang to DESCRIPTION

christopher-mohr committed:
clean up, fix issues

christopher-mohr committed:
fix issues with df column names

christopher-mohr committed:
fix further warnings

christopher-mohr committed:
more code and docs improvements

christopher-mohr committed:
update NAMESPACE

christopher-mohr committed:
fix warnings

christopher-mohr committed:
fix descriptions

christopher-mohr committed:
update docs

christopher-mohr committed:
add missing param descriptions

christopher-mohr committed:
resolve warnings caused by documentation

christopher-mohr committed:
still some missing descriptions

christopher-mohr committed:
add missing doc file

apeltzer committed:
Merge pull request #35 from christopher-mohr/ci_doc_website

Add CI and build/improve docs

apeltzer committed:
Add ubuntu-latest for building docs

apeltzer committed:
Trying to fix CI for pkgdown

apeltzer committed:
Merge pull request #37 from Boehringer-Ingelheim/improve-ci

Improve CI

christopher-mohr committed:
improve docs

christopher-mohr committed:
Merge remote-tracking branch 'origin/main' into final_touches_docs

christopher-mohr committed:
update docs workflow

christopher-mohr committed:
docs, workflow, pages

christopher-mohr committed:
remove roxygenize workflow

christopher-mohr committed:
fix mac os test

christopher-mohr committed:
fallback for mac os tests

christopher-mohr committed:
continue on error for macos

apeltzer committed:
Merge pull request #38 from christopher-mohr/final_touches_docs

Yeah now that the 2 packages are indexed properly, tests run fine 👍🏻

grst committed:
Don't build docs on release events

Should be enough to build it on `main`

christopher-mohr committed:
Merge pull request #39 from Boehringer-Ingelheim/grst-patch-2

Don't build docs on release events
